addons/crm/views/crm_lead_views.xml

```diff
@@ -864,7 +864,9 @@
             <field name="binding_view_types">list</field>
             <field name="state">code</field>
             <field name="code">
-if record:
+if not 'opportunity' in record.mapped('type'):
+    record.action_set_lost()
```

Comment:
I think code server action are always run once for multiple records, so this should be:

```suggestion
if not 'opportunity' in records.mapped('type'):
    records.action_set_lost()
```
to not just modify the first record.